docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+| transactionHash | bytes | | empty |
+| consensusTimestamp | Timestamp | | The consensus timestamp of the autorenewal |
+| transactionID | TransactionID | | { empty transactionValidStart, autoRenewAccount } |
+| memo | string | | "Entity {ID} was automatically renewed. New expiry: {newExpiry}" |
+| transactionFee | uint64 | | The fee charged for the autorenewal of the entity |
+| contractCallResult | ContractFunctionResult | | empty |
+| contractCreateResult | ContractFunctionResult | | empty |
+| transferList | TransferList | | {(autoRenewAccount, -transactionFee), (defaultFeeCollectionAccount, transactionFee)} |
+| tokenTransferLists | TokenTransferList | repeated | empty |
+| scheduleRef | ScheduleID | | empty |
+
+The main difference between an autorenewal record and a regular `TransactionRecord` is that it has an empty `transactionHash` and an empty `transactionID.transactionValidStart`. There were older versions of Hedera Services that did not have `transactionHash` in a `TransactionRecord`, but an empty `transactionID.transactionValidStart` will guarantee that the `TransactionRecord` was generated in place of an autorenewal record.
+
+## Autodeletion record
+After autodeleting an entity due to the zero balance of the `autoRenewAccount`, Hedera Services will generate a `TransactionRecord` that serves as an autodeletion record and contains the following:
+
+| Field | Type | Label | Description |
+|---|---|---|---|
+| receipt | TransactionReceipt | | receipt will contain either an accountID, a fileID, a contractID, a topicID or a tokenID that got autodeleted |
+| transactionHash | bytes | | empty |
+| consensusTimestamp | Timestamp | | The consensus timestamp of the autodeletion |
+| transactionID | TransactionID | | { empty transactionValidStart, autoRenewAccount } |
+| memo | string | | "Entity {ID} was automatically deleted." |
+| transactionFee | uint64 | | 0 |
+| contractCallResult | ContractFunctionResult | | empty |
+| contractCreateResult | ContractFunctionResult | | empty |
+| transferList | TransferList | | empty |
+| tokenTransferLists | TokenTransferList | repeated | empty |
+| scheduleRef | ScheduleID | | empty |
+
+An autodeletion record will be very similar to an autorenewal record. The `memo` will reflect that the entity was automatically deleted. Due to the zero balance of the `autoRenewAccount`, the `transactionFee` is zero and the `transferList` is empty.
+
+## Special notes
+- At the time of this writing, a file is not associated with an `autoRenewAccount` so a file can only be renewed by a fileUpdate transaction.
+- Because cryptoDelete transfers all remaining balance of an account to another account, all accounts that are marked `deleted` will have zero balance. In other words, an account with a non-zero balance will never be marked `deleted`. Step 2 of the implementation does not need to worry about the `autoRenewAccount` being marked `deleted` before.
```

Comment:
A crypto account has HBAR's. However, a token entity doesn't implicitly have tokens.
Is there a provision to allow token holders to renew the token? Does this role fall to the treasury account?

Also, should a token need to expire what happens to the associated tokens in treasury account and all associated accounts?
Do they just remain and all attempted transfers fail with `TOKEN_DELETED` or is there provision for other accounts  to pay for renewal to ensure the mini token economy continues?